src/jvm/test/aws/sdk/kotlin/crt/http/HttpRequestResponseTest.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.sdk.kotlin.crt.http
+
+import aws.sdk.kotlin.crt.runSuspendTest
+import aws.sdk.kotlin.crt.util.Digest
+import aws.sdk.kotlin.crt.util.encodeToHex
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.TestInstance
+import org.mockserver.client.MockServerClient
+import org.mockserver.integration.ClientAndServer
+import org.mockserver.model.HttpRequest.request
+import org.mockserver.model.HttpResponse.response
+import kotlin.test.*
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class HttpRequestResponseTest : HttpClientTest() {
+    private val TEST_DOC_LINE =
+        "This is a sample to prove that http downloads and uploads work. It doesn't really matter what's in here, we mainly just need to verify the downloads and uploads work."
+    private val TEST_DOC_SHA256 = "c7fdb5314b9742467b16bd5ea2f8012190b5e2c44a005f7984f89aab58219534"
+
+    var mockServer: MockServerClient = MockServerClient("localhost", 0)
+    var url = "http://localhost:"
+
+    @BeforeAll
+    fun setup() {
+        mockServer = ClientAndServer.startClientAndServer(0)
+        url += mockServer.port
+    }
```

Comment:
**Nit**: Why set `mockServer` and `url` in field initializers and then re-set them in `setup`? Seems like a perfect case for `lateinit`:

```kotlin
lateinit var mockServer: MockServerClient
lateinit var url: String

@BeforeAll
fun setup() {
    mockServer = ClientAndServer.startClientAndServer(0)
    url = "http://localhost:${mockServer.port}"
}
```